frontend/components/admin.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import io from 'socket.io-client';
+import styled from 'styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { fas } from '@fortawesome/free-solid-svg-icons';
+
+library.add(fas);
+
+
+const CheckIcon = styled(FontAwesomeIcon).attrs( props => ({
+  icon: props.solved ? 'check' : 'times',
+}))`
+  color: ${props => (props.solved ? 'green' : 'red')};
+`;
+class Admin extends React.Component {
+  constructor(props) {
+    super(props);
+    const socket = io(SOCKET_URL);
+    socket.on('admin_return', (data) => {
+      this.setState({ gameState : data['gameState'], tableData : data['tableData'] });
+    });
+    this.state = {
+      socket:socket,
+      solved:false,
+      tableData : [['initialValue']],
+      gameState : [''],
+      info: [['Hello! Welcome to the Admin Page. Press the button to see the progress!']],
+    };
+    this.adminPing();
+  }
+
+  adminPing() {
+    const { socket } = this.state;
+    socket.emit('admin_ping', {});
+  }
+
+  startGame() {
+    console.log('this will be the start game button');
+  }
+  renderClients(table) {
+    const { tableData } = this.state;
+    const clientElements = Object.entries(tableData[table]).map((elements,i) => {
+      var clients = elements[1];
+      return (
+        <li key={i}>{clients}</li>
+      );
+    });
+    return clientElements;
+  }
+
+  renderPuzzle(table) {
+    const { gameState } = this.state;
+    const puzzleElements = Object.entries(gameState[table]).map(elements => {
+      var puzzle = elements[0];
+      var solved = elements[1]['solved'];
+      return (
+        <li key={puzzle}>puzzle {puzzle}: <CheckIcon solved={solved}></CheckIcon></li>
```

Comment:
this can be simplified to `<CheckIcon solved={solved} />` since there's no children